controller/tap/server.go

```diff
@@ -468,20 +470,29 @@ func NewServer(
 		return nil, nil, err
 	}
 
-	s, _ := newGRPCTapServer(tapPort, controllerNamespace, k8sAPI)
+	globalConfig, err := config.Global(pkgK8s.MountPathGlobalConfig)
+	clusterDomain := globalConfig.GetClusterDomain()
+	if err != nil || clusterDomain == "" {
```

Comment:
Likewise, I think we should handle the two conditional here separately. The `log.Warnf` is probably not needed since we are falling back to `cluster.local` anyway (i.e. what can the user do with the information?).

Reply:
The information was valuable to check if your tap server actually uses the cluster domain you configured it with. 

That being said. I also like your suggestion of splitting the the errors and printing an info log about which cluster domain is being used.